codec/src/main/java/io/netty/handler/codec/compression/CompressionCodecFactory.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import io.netty.util.internal.ObjectUtil;
+
+/**
+ * Creates a new {@link CompressionEncoder} or a new {@link CompressionDecoder}
+ * with specified {@link CompressionFormat}.
+ */
+public final class CompressionCodecFactory {
+
+    /**
+     * Factory method for a new {@link CompressionEncoder}.
+     *
+     * @param format specified compression format.
+     */
+    public static CompressionEncoder newEncoder(CompressionFormat format) {
+        ObjectUtil.checkNotNull(format, "format");
+        switch (format) {
+            case ZLIB:
+                return ZlibCodecFactory.newZlibEncoder(ZlibWrapper.ZLIB);
+            case GZIP:
+                return ZlibCodecFactory.newZlibEncoder(ZlibWrapper.GZIP);
+            case DEFLATE:
+                return ZlibCodecFactory.newZlibEncoder(ZlibWrapper.NONE);
+            case ZLIB_OR_NONE:
+                throw new UnsupportedOperationException("In case of " + ZlibWrapper.ZLIB_OR_NONE
+                        + " an encoder can not be auto-detected");
```

Comment:
Let's use `IllegalArgumentException` and use the following message:

    "format: " + format + " (" + ZlibWrapper.ZLIB_OR_NONE + " is not allowed for an encoder + ')'